create/application_test.go

```diff
@@ -47,10 +51,18 @@ func TestApplication(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	authSecret := gitAuthSecret(cmd.Git, app.Name, app.Namespace)
+	if err := apiClient.Get(ctx, api.ObjectName(authSecret), authSecret); err != nil {
+		t.Fatal(err)
+	}
+
 	assert.Equal(t, cmd.Name, app.Name)
-	assert.Equal(t, cmd.GitURL, app.Spec.ForProvider.Git.URL)
-	assert.Equal(t, cmd.GitSubPath, app.Spec.ForProvider.Git.SubPath)
-	assert.Equal(t, cmd.GitRevision, app.Spec.ForProvider.Git.Revision)
+	assert.Equal(t, cmd.Git.URL, app.Spec.ForProvider.Git.URL)
+	assert.Equal(t, cmd.Git.SubPath, app.Spec.ForProvider.Git.SubPath)
+	assert.Equal(t, cmd.Git.Revision, app.Spec.ForProvider.Git.Revision)
+	assert.Equal(t, cmd.Git.Username, string(authSecret.Data["username"]))
+	assert.Equal(t, cmd.Git.Password, string(authSecret.Data["password"]))
+	assert.Equal(t, cmd.Git.SSHPrivateKey, app.Spec.ForProvider.Git.Auth.SSHPrivateKey)
```

Comment:
this line doesn't test anything now. We should also test secret data here, i.e. similar to the un and pw:
```
assert.Equal(t, cmd.Git.SSHPrivateKey, string(authSecret.Data["privatekey"]))
```
of course we should also set some test data for the `privatekey`

Reply:
Good catch. Though we need to create a separate app to test that since the user/pass and ssh key are mutually exclusive.

Reply:
looks good 👌